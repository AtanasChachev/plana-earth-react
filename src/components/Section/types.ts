export interface SectionProps {
  children: JSX.Element | JSX.Element[];
  isFullHeight?: boolean;
  headerTitle?: string;
  shAlignInlineBlockOnMobile?: boolean;
  isEmptyBlockActive: boolean;
}