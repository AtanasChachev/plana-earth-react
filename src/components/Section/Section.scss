@import 'styles/utils/mixins';

.section {
  padding: 20px;
  
  &__inner {
    position: relative;
    background-color: var(--clr-white);
    box-shadow: 0 0 16px 16px var(--box-shadow);
    border-radius: 4px;

    &--full-height {
      @include breakpoint(min, width, 960px) {
        height: calc(100vh - 40px);
        overflow: auto;
      }
    }

    &__header {
      padding: 14px 10px;
      border-radius: 4px 4px 0 0;
      background-color: var(--clr-white-ice);

      &__title {
        font-size: 14px;
        margin: 0;
        letter-spacing: 0.5px;
        font-weight: 300;

        @include breakpoint(min, width, 960px) {
          font-size: 18px;
        }
      }
    }

    &__content {
      padding: 10px;

      @include breakpoint(min, width, 960px) {
        padding: 20px;
      }

      &--mobile-inline-blocks {
        @include breakpoint(max, width, 959px) {
          display: flex;
          flex-wrap: wrap;

          > div {
            flex: 1 1 calc(50% - 20px);

            &:nth-child(2n + 1) {
              margin-right: 10px;
            }

            &:nth-child(2n + 2) {
              margin-left: 10px;
            }
          }
        }
      }
    }
  }
}