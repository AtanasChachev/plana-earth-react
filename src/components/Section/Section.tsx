import { Loader } from 'components/index';
import './Section.scss';

type SectionProps = {
  children: JSX.Element | JSX.Element[];
  isFullHeight?: boolean;
  headerTitle?: string;
  shRenderLoader?: boolean;
  shAlignInlineBlockOnMobile?: boolean;
  isEmptyBlockActive: boolean;
};

const Section = ({ children, isFullHeight, headerTitle, shAlignInlineBlockOnMobile, shRenderLoader, isEmptyBlockActive }: SectionProps): JSX.Element => {
  return (
    <div className="section">
      <div className={`section__inner ${isFullHeight ? 'section__inner--full-height' : ''}`}>
        {
          headerTitle && headerTitle.length && <>
            <div className="section__inner__header">
              <h3 className="section__inner__header__title">{ headerTitle }</h3>
            </div>
          </>
        }

        <div className={`section__inner__content ${shAlignInlineBlockOnMobile ? 'section__inner__content--mobile-inline-blocks' : ''} ${isEmptyBlockActive ? 'section__inner__content--center' : ''}`}>
          {children}
        </div>

        {
          shRenderLoader && <Loader />
        }
      </div>
    </div>
  );
};

export { Section };
