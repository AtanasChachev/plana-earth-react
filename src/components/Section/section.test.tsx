import { render } from '@testing-library/react';
import Section from './Section';

const setup = () => {
  const { asFragment } = render(
    <Section
      children={<div>Test message</div>}
      isFullHeight={true}
      headerTitle='Test header title'
      shAlignInlineBlockOnMobile={true}
      shRenderLoader={false}
      isEmptyBlockActive={false}
    />,
  );

  return {
    asFragment,
  };
};

describe('<Section />', () => {
  test('<Section /> renders correctly', () => {
    const { asFragment } = setup();

    expect(asFragment()).toMatchSnapshot();
  });
});