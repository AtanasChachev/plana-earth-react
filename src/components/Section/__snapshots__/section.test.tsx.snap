// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Section /> <Section /> renders correctly 1`] = `
<DocumentFragment>
  <div
    class="section"
  >
    <div
      class="section__inner section__inner--full-height"
    >
      <div
        class="section__inner__header"
      >
        <h3
          class="section__inner__header__title"
        >
          Test header title
        </h3>
      </div>
      <div
        class="section__inner__content section__inner__content--mobile-inline-blocks "
      >
        <div>
          Test message
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
